app/controllers/polls_controller.rb

```diff
@@ -0,0 +1,43 @@
+class PollsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :initialize_redirect_path
+
+  def feeds
+    respond_to do |format|
+      format.html {
+        @feeds_poll_status_path = feeds_poll_status_path
+        unless allowable_redirect_path?
+          flash.now[:notice] = "Oops! I'm not sure where to go from here, are you a bad guy?"
+          # TODO: Log any other information we think may be helpful here e.g. IP, etc.
+          Rails.logger.error("redirect_path was not in the allowed list: #{@redirect_path}")
+        end
+      }
+    end
+  end
+
+  def feeds_poll
+    data = {
+      running: helpers.feed_update_running?,
+    }
+    render json: data.to_json, status: :ok
+  end
+
+  private
+
+  def initialize_redirect_path
+    @redirect_path = params[:redirect_path]
+  end
+
+  def allowable_redirect_path?
+    allowable_redirect_paths.include? @redirect_path
+  end
+
+  # This would protect against something like a phishing scam, where phishers would send
+  # an email with a link http://www.domain.com/polls/feeds?redirect_path=http://malicious.com.
+  # The unsuspecting user would first be asked to authenticate, then, redirected to http://malicious.domain.com
+  # after authentication, where bad stuff can happen. We could push/pop the redirect_url from session, but
+  # this is fine for our code challenge.
+  def allowable_redirect_paths
+    @allowable_redirect_paths ||= [root_path, recommended_news_path]
+  end
+end
```

Comment:
If we're viewing feeds (news items) or recommended feeds (news items "starred" by users) and the PullFeedsJob is running, the user is redirected here#feeds, where the PullFeedsJob is polled until it is complete; then, the user is redirected back to the view they were previously on. Redirect urls are checked against an "allowable" list to avoid redirect_path redirecting to a malicious web site (e.g. phishing scam)